src/main/java/io/kestra/plugin/aws/athena/Query.java

```diff
@@ -0,0 +1,346 @@
+                default -> Thread.sleep(500);
+            }
+        }
+        while (queryExecution.status().state() != QueryExecutionState.SUCCEEDED);
+
+        return queryExecution != null ? queryExecution.statistics() : null;
+    }
+
+    private Map<String, Object> fetchOne(List<ColumnInfo> columnInfo, List<Row> results) {
+        if (results == null || results.isEmpty()) {
+            return null;
+        }
+
+        Row row = results.get(0);
+        return map(columnInfo, row);
+    }
+
+    private List<Object> fetch(List<ColumnInfo> columnInfo, List<Row> results) {
+        if (results == null || results.isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        return results.stream().map(row -> (Object) map(columnInfo, row)).toList();
+    }
+
+    private Pair<URI, Long> store(List<ColumnInfo> columnInfo, List<Row> results, RunContext runContext) throws IOException {
+        if (results == null || results.isEmpty()) {
+            return Pair.of(null, 0L);
+        }
+
+        File tempFile = runContext.tempFile(".ion").toFile();
+        AtomicLong count = new AtomicLong();
+
+        try (var output = new FileOutputStream(tempFile)) {
+            results.forEach(throwConsumer(row -> {
+                count.incrementAndGet();
+                FileSerde.write(output, map(columnInfo, row));
+            }));
+        }
+
+        return Pair.of(
+            runContext.putTempFile(tempFile),
+            count.get()
+        );
+    }
+
+    private Map<String, Object> map(List<ColumnInfo> columnInfo, Row row) {
+        if (!row.hasData()) {
+            return null;
+        }
+
+        Map<String, Object> data = new HashMap<>();
+        for (int i = 0; i < columnInfo.size(); i++) {
+            data.put(columnInfo.get(i).name(), mapCell(columnInfo.get(i), row.data().get(i)));
+        }
+        return data;
+    }
+
+    private Object mapCell(ColumnInfo columnInfo, Datum datum) {
+        // We try our best to convert the result to a precise type as all data comes as a varchar.
+        // See https://docs.aws.amazon.com/athena/latest/ug/data-types.html for the list of supported types.
+        return switch (columnInfo.type()) {
+            case "boolean" -> Boolean.valueOf(datum.varCharValue());
+            case "tinyint", "smallint", "int", "integer" -> Integer.valueOf(datum.varCharValue());
+            case "bigint" -> Long.valueOf(datum.varCharValue());
+            case "double" -> Double.valueOf(datum.varCharValue());
+            case "float" -> Float.valueOf(datum.varCharValue());
+            case "decimal" -> new BigDecimal(datum.varCharValue());
+            case "date" -> LocalDate.parse(datum.varCharValue(), dateFormatter);
+            case "timestamp" -> LocalDateTime.parse(datum.varCharValue(), timestampFormatter);
+            // default correspond to the types char, varchar, string, binary, array, map, struct
+            default -> datum.varCharValue();
```

Comment:
seems that we need to cover more type here like map ...

Reply:
This is excluded and documented as is, the issue is that maps and arrays can be typed so we also need to parse the type field if we want to convert arrays and maps.
And it's not easy to test as CSV will not produce this kind of structures.
So it will need a lot of work, my plan is to ship as is and implements it later if someone request it, in this case we will b able to ask for a reproducer to understand how complex types are handled by Athena.